### PR TITLE
Add script to generate BIRD6 mesh config

### PR DESCRIPTION
**Pull request by neiljerram**

### CODE REVIEW

File: etc/calico-gen-bird6-mesh-conf.sh [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    BIRD_CONF=/etc/bird6.conf
+else
+    BIRD_CONF=/etc/bird/bird6.conf
+fi
+
+TEMPLATE_DIR=${TEMPLATE_DIR:-/usr/share/calico/bird}
+BIRD_CONF_TEMPLATE=${TEMPLATE_DIR}/calico-bird6.conf.template
+BIRD_CONF_PEER_TEMPLATE=${TEMPLATE_DIR}/calico-bird6-peer.conf.template
+
+# Require at least 4 arguments.
```
Comment (fasaxc) [inline etc/calico-gen-bird6-mesh-conf.sh]: Line below says 3, is that correct?

File: etc/calico-gen-bird6-mesh-conf.sh [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+where
+  <my-ipv4-address> is the external IPv4 address of the local machine
+  <my-ipv6-address> is the external IPv6 address of the local machine
+  <as-number> is the BGP AS number that we should use
+  each <peer-ipv6-address> is the IPv6 address of another BGP speaker that
+      the local BIRD should peer with.
+
+EOF
+[ $# -ge 3 ] || exit -1
```
Comment (fasaxc) [inline etc/calico-gen-bird6-mesh-conf.sh]: Aren't exit statuses normally positive?

**Comment by fasaxc** [comment]
Just a couple of minor questions, otherwise looks good to me.

File: etc/calico-gen-bird6-mesh-conf.sh (hunk 2 above)
Comment (neiljerram) [inline etc/calico-gen-bird6-mesh-conf.sh]: I think I've seen negative statuses generally used.  -1 becomes 255:

    neil@nj-ubuntu:~/calico/networking-calico$ bash -c 'exit -1'
    neil@nj-ubuntu:~/calico/networking-calico$ echo $? 
    255

Anything that's non-0 is equally false.

File: etc/calico-gen-bird6-mesh-conf.sh (hunk 1 above)
Comment (neiljerram) [inline etc/calico-gen-bird6-mesh-conf.sh]: The code is correct - I'll update the comment!  (Here and in the IPv4 script.)

**Comment by neiljerram** [comment]
Agreed with @fasaxc that we can merge this now...

**Pull request merged**